src/pages/workspace/distanceRates/PolicyDistanceRatesPage.tsx

```diff
@@ -0,0 +1,283 @@
+import Icon from '@components/Icon';
+import * as Expensicons from '@components/Icon/Expensicons';
+import * as Illustrations from '@components/Icon/Illustrations';
+import Modal from '@components/Modal';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import TableListItem from '@components/SelectionList/TableListItem';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import type {CentralPaneNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import {openPolicyDistanceRatesPage} from '@userActions/Policy';
+import ButtonWithDropdownMenu from '@src/components/ButtonWithDropdownMenu';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type Policy from '@src/types/onyx/Policy';
+import type {CustomUnit, Rate} from '@src/types/onyx/Policy';
+
+type RateForList = {
+    value: string;
+    text: string;
+    keyForList: string;
+    isSelected: boolean;
+    rightElement: React.ReactNode;
+};
+
+type PolicyDistanceRatesPageOnyxProps = {
+    /** Policy details */
+    policy: OnyxEntry<Policy>;
+};
+
+type PolicyDistanceRatesPageProps = PolicyDistanceRatesPageOnyxProps & StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.WORKSPACE.DISTANCE_RATES>;
+
+function PolicyDistanceRatesPage({policy, route}: PolicyDistanceRatesPageProps) {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const {translate} = useLocalize();
+    const [selectedDistanceRates, setSelectedDistanceRates] = useState<Rate[]>([]);
+    const [isWarningModalVisible, setIsWarningModalVisible] = useState(false);
+    const dropdownButtonRef = useRef(null);
+
+    const customUnit: CustomUnit | undefined = useMemo(
+        () => (policy?.customUnits !== undefined ? policy?.customUnits[Object.keys(policy?.customUnits)[0]] : undefined),
+        [policy?.customUnits],
+    );
+    const customUnitRates: Record<string, Rate> = useMemo(() => customUnit?.rates ?? {}, [customUnit]);
+
+    useEffect(() => {
+        openPolicyDistanceRatesPage(route.params.policyID);
+    }, [route.params.policyID]);
+
+    const distanceRatesList = useMemo<RateForList[]>(
+        () =>
+            Object.values(customUnitRates).map((value) => ({
+                value: value.customUnitRateID ?? '',
+                text: `${CurrencyUtils.convertAmountToDisplayString(value.rate, value.currency ?? CONST.CURRENCY.USD)} / ${translate(
```

Comment:
@luacmartins This is something that Im not sure of. I've done this bit similar to existing rates that are in the app. Is this correct? Should I use exactly this method to display the rate that comes from API?

Reply:
yes, this is correct